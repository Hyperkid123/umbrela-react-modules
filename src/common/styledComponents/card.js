import styled from 'styled-components';
import grey from 'material-ui/colors/grey';
import blue from 'material-ui/colors/blue';
import red from 'material-ui/colors/red';

export const CardWrapper = styled.div`
  position: relative;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: ${props => props.questionList ? '700px' : 0};
  max-width: ${props => props.questionList ? props => props.matrix ? 'auto' : '700px' : 'auto'};
  word-wrap: break-word;
  background-color: ${props => props.transparent ? `transparent` : `#fff`};
  background-clip: border-box;
  box-sizing: border-box;
  border: 0;
  box-shadow: ${props => props.transparent ? `none` : `0 2px 5px 0 rgba(0,0,0,.16), 0 2px 10px 0 rgba(0,0,0,.12)`};
  border-radius: 2px!important;
  margin-bottom: .5rem!important;
  font-weight: 400;
  margin-left: 15px;
  margin-right: 15px;
`

export const CardHeader = styled.div`
  color: #212121 !important;
  min-width: 0;
  white-space: ${props => props.questionList ? 'wrap' : 'nowrap'};
  overflow:${props => props.questionList ? 'overflow' : 'hidden'};
  text-overflow:${props => props.questionList ? 'initial' : 'ellipsis'};
  padding: .75rem 1.25rem;
  margin-bottom: 0;
  background-color: ${props => !props.transparent ? `rgba(0,0,0,.03)` : `rgba(255,255,255,.8)`};
  border-bottom: 1px solid;
  border-color: #BDBDBD;
  box-sizing: border-box;
  box-shadow: ${props => !props.transparent ? `none` : `0 2px 5px 0 rgba(0,0,0,.16), 0 2px 10px 0 rgba(0,0,0,.12)`};
  h2 {
    font-weight: 400;
    margin: .25em 0;
  }
`

export const CardBody = styled.div`
  position: relative;
  -webkit-box-flex: 1;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: 1.25rem;
  box-sizing: border-box;
  border-top: ${props => props.questionList ? `10px solid` : `none`};
  border-color: ${props => props.error ? red.A200 : 'transparent'};
`
export const InputHeader = styled.div`
  padding-top: 12px;
  background-color: rgba(0, 0, 0, 0.03);
  height: 49px;
  box-sizing: border-box;
  position: relative;
  svg {
    padding-left: 8px;
    position: absolute;
    top: 0;
    color: ${grey[700]};
  }
  input {
    margin: 0;
    padding: 1px 20px 12px 28px;
    width: calc(100% - 40px);
    color: #212121 !important;
    height: auto;
  }
`

export const CardControlls = styled.div`
  padding-left: 1.25rem;
  padding-right: 1.25rem;
`
export const CardListItemheading = styled.h3`
  font-weight: 400;
  margin: .2em 0;
  span:first-child {
    display: inline-block;
    height: 1.25em;
    width: 1.25em;
    background-color: ${blue[500]};
    color: white;
    padding: .25em;
    text-align: center;
    border-radius: 2px;
    margin-right: .25em;
  }
`
