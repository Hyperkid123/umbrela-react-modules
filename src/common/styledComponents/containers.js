import styled from 'styled-components';
import {
  white,
  grey300, grey500, grey700, grey200, grey100,
  red500,
} from 'material-ui/styles/colors';

export const Flex = styled.div`
  width: ${props => props.auto ? 'auto' : '100%'};
  max-height: 100%;
  display: flex;
  flex-direction: ${props => props.column ? 'column' : 'row'};
  align-items:  ${props => props.verticalCenter ? 'center' : 'null'};
  align-items: ${props => props.baseline ? 'baseline' : 'null'};
  justify-content: ${props => props.horizintalCenter ? 'center' : 'null'};
  justify-content: ${props => props.end ? 'flex-end' : 'null'};
  flex-grow: ${props => props.grow ? 1 : 0};
  flex-wrap: ${props => props.wrap ? 'wrap' : 'no-wrap'};
`

export const ViewContainer = styled.div`
  width: calc(100% - 2em);
  min-height: calc(100vh - 1em - 28px);
  margin: 0;
  padding: 0 1em;
`

export const MenuContainer = styled.div`
  max-width: ${props => props.fullWidth ? '100&' : '400px'};
  background-color: ${white};
  padding: 10px;
`

export const FlexSection = styled.section`
  margin: 10px;
  min-width: ${props => props.minWidth ? `${props.minWidth}px` : `0px`};
  height: ${props => props.autoHeight ? 'auto' :  'calc(100vh - 2em - 20px - 28px)'};
  width: ${props => props.fullWidth ? '100%' : 'auto'};
`

export const MenuList = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
  margin-right: 8px;
`
export const MenuItem = styled.div`
  padding: 0 .5em 0 .5em;
  font-size: 1.2em;
  height: 48px;
  cursor: pointer;
  transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  background-color: ${props => props.active ? grey300 : white};
  &:hover {
    background-color: ${props => !props.dragging ? grey300 : white};
  }
  > div {
    height: 100%;
  }
`

export const MenuListIconsContainer = Flex.extend`
  display: none;
  margin-left: auto;
  width: auto;
`

export const MenuListLabel = styled.div`
  max-width: 100%;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
`
export const LabelContainer = styled.div`
  margin-left: ${props => props.alignRight ? 'auto' : 0};
  padding-left: 0;
  padding-right: 15px;
  font-size: 0.8em;
  color: ${props => props.error ? red500 : grey500};
`

export const PreviewImage = styled.img`
  max-width: 600px;
  height: auto;
  border-radius: 5px;
`
export const FullImage = PreviewImage.extend`
  max-width: 90%;
  display: block;
  margin-left: auto;
  margin-right:auto;
`

export const OptionsList = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
`

export const OptionItemWrapper = Flex.extend`
  > div:last-of-type, button:last-of-type{
    display: none !important;
  }
  &:hover{
    > div:last-of-type{
      display: block !important;
    }
    > button:last-of-type {
      display: block !important;
    }
  }
`

export const CustomHelpWrapper = styled.div`
  font-size: .9em;
  color: ${grey700}
  margin: 0;
  padding: 10px;
`

export const CustomHelpLine = styled.div`
  margin:0 0 5px 0;
  padding: 0;
  &:last-child {
    margin: 0;
  }
`
export const OptionsFillListWrapper = Flex.extend`
  padding-left: 20px;
  padding-right: 20px;
  width: calc(100% - 40px);
`

export const PreviewImageContainer = styled.div`
  display: table;
  margin-right: 1em;
`
export const OrderQuestionList = styled.div`
  width: 100%;
`

export const OrderQuestionItem = styled.div`
  width: 420px;
  margin-left: auto;
  margin-right: auto;
  padding: 1em 1rem;
  padding-left: .2em;
  margin-bottom: .5rem;
  background-color: ${grey200};
  cursor: move;
  display: flex;
  align-items: center;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  box-shadow: rgba(0, 0, 0, 0.12) 0px 1px 6px, rgba(0, 0, 0, 0.12) 0px 1px 4px;
  > svg{
    opacity: 0;
  }
  &:hover > svg {
    opacity: 1;
  }
`
export const ListColumn = styled.div`
  padding-left: 20px;
  padding-right: 20px;
`
export const MatrixTable = styled.table`
  border-collapse:collapse;
`
export const MatrixTableHeader = styled.thead`
  text-align: center;
`
export const MatrixtableCell = styled.td`
  padding: .4em;
  text-align: center;
`
export const MatrixTableBody = styled.tbody`

`

export const MatrixTableRow = styled.tr`
  &:nth-child(odd) > td:not(:first-of-type){
    background-color: ${props => props.filler ? grey200 : white}
  }
  > td:not(:first-of-type){
    background-color: ${props => props.filler ? grey100 : white}
  }
`

export const SectionBody = styled.div`
  max-height: ${props => props.open ? '1500px' : 0};
  transition: max-height .5s ease-in;
  overflow: hidden;
  height:auto;
  >div{
    padding: 5px;
  }
`;

export const Sectionhead = styled.div`
  cursor: pointer;
  padding: 10px;
`;

export const PyramidTile = styled.div`
  width: 96px;
  height: 96px;
  margin: 2px
`

export const AddPyramidTile = styled.div`
  display: flex;
  flex-shrink: 10;
  padding: 5px;
  justify-content: center;
  align-items: center;
  min-width: 100px;
`
