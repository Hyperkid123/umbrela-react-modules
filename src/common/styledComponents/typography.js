import styled from 'styled-components';
import {
  deepOrange500,
  grey800, grey900,
  black,
  white
} from 'material-ui/styles/colors';

const BaseHeading = styled.h2`
  font-weight: 300;
  border-bottom: 1px solid ${deepOrange500};
  border: ${props => props.withouthBorder ? 'none' : ''};
  color: ${grey800};
`;

export const BigHeading = BaseHeading.extend`
  font-size: 56px;
`
export const MediumHeading = BaseHeading.extend`
  font-size: 34px;
  margin-top: 10px;
  margin-bottom: 10px;
`
export const SmallHeading = BaseHeading.extend`
  font-size: 24px;
  margin-top: 5px;
  margin-bottom: 5px;
`

export const DefaultLink = styled.a`
  color: ${black};
  text-decoration: none;
`

export const FillQuestionheading = SmallHeading.extend`
  border: none;
  font-weight: 500;
  ::before {
    content: counter(step-increment);
    font-weight: bold;
    padding: 3px 8px;
    margin-right: 3px;
    float: left;
    background-color: ${deepOrange500};
    color: ${white};
    border-radius: 2px;
    font-size: 0.8em;
  }
`

export const CustomHelp = styled.span`
  font-size: 14px;
  margin-top: 5px;
  margin-bottom: 5px;
  color: ${grey800}
`
export const DeleteNotification = styled.span`
  font-style:italic;
  color: ${grey900}
`
export const StatusMessage = styled.span`
  position: relative;
  bottom: 5px;
`
