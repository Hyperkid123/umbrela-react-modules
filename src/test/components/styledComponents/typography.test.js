import React from 'react'
import {SmallHeading} from '../../../common/styledComponents/typography';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Typography styled components', () => {
  it('should render heading withouth border', () => {

    const tree = shallow(
      <SmallHeading>
        foo
      </SmallHeading>
    )
    expect(toJson(tree)).toMatchSnapshot()

    const noBorderTree = shallow(
      <SmallHeading withouthBorder>
        foo
      </SmallHeading>
    )
    expect(toJson(noBorderTree)).toMatchSnapshot()
  })
});
