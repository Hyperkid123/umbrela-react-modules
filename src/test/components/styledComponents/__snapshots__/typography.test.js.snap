// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography styled components should render heading withouth border 1`] = `
<h2
  className="sc-htpNat dOuKMx"
>
  foo
</h2>
`;

exports[`Typography styled components should render heading withouth border 2`] = `
<h2
  className="sc-htpNat hdEPeJ"
>
  foo
</h2>
`;
